gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java

```diff
@@ -92,6 +102,13 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
             try {
                 authenticator.authenticate(credentials);
                 ctx.fireChannelRead(request);
+
+                // User name logged with the remote socket address and authenticator classname for audit logging
+                if (authenticationSettings.enableAuditLog) {
+                    String[] authClassParts = authenticator.getClass().toString().split("[.]");
+                    auditLogger.info("User {} with address {} authenticated by {}", credentials.get(PROPERTY_USERNAME),
+                            ctx.channel().remoteAddress().toString().substring(1), authClassParts[authClassParts.length - 1]);
```

Comment:
It looks suspect, indeed, thanks for the remark. remoteAddres is a SocketAddress though, so it would not fail on address resolution. To keep remoteAddress use consistent with gremlin-driver and be on the conservative side, I do not mind putting the string operation sequence in a try{} block (also for the other occurrences below).